hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/source/ScanContext.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.source;
+
+import org.apache.hudi.configuration.FlinkOptions;
+
+import org.apache.flink.annotation.Internal;
+import org.apache.flink.configuration.Configuration;
+import org.apache.flink.core.fs.Path;
+import org.apache.flink.table.types.logical.RowType;
+
+import java.io.Serializable;
+import java.time.Duration;
+
+/**
+ * Hudi source scan context.
+ */
+@Internal
+public class ScanContext implements Serializable {
+  private final Configuration conf;
+  private final Path path;
+  private final RowType rowType;
+  private final String startInstant;
+  private final long maxCompactionMemoryInBytes;
+  private final long maxPendingSplits;
+  // skip compaction
+  private final boolean skipCompaction;
+  // skip clustering
+  private final boolean skipClustering;
+  // skip insert overwrite
+  private final boolean skipInsertOverwrite;
+  // cdc enabled
+  private final boolean cdcEnabled;
```

Comment:
I didn't see the endInstant and rangeType to describe the read boundary specifically.

Reply:
Actually, both IncrementalQueryAnalyzer IncrementalInputSplits has duplicated fields. As we discussed offline, I feel the code can be simplified to extract the common parameters for both IncrementalInputSplits and IncrementalQueryAnalyzer. How do you think?

Reply:
I will do the refactor together with resolving the issue #17606